### PR TITLE
SONAR-6701 - Drop the db table SEMAPHORES

### PR DESCRIPTION
**Pull request by simonbrandhof**
SONAR-6701 - Drop the db table SEMAPHORES

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/semaphore/SemaphoresImpl.java [hunk 1]
```diff
@@ -21,34 +21,24 @@
...
   }
 
   @Override
   public void release(String name) {
-    updater.stopUpdate(name);
-    dao.release(name);
+    throw fail();
+  }
+
+  private RuntimeException fail() {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/semaphore/SemaphoresImpl.java]: MAJOR Make "fail" a "static" method. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)

File: sonar-plugin-api/src/main/java/org/sonar/api/utils/Semaphores.java [hunk 2]
```diff
@@ -27,8 +27,10 @@
  * A semaphore shared among all the processes that can connect to the central database.
  *
  * @since 3.4
+ * @deprecated in 5.2 as batch analyzer does not connect to database anymore
  */
 @ServerSide
+@Deprecated
 public interface Semaphores {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/utils/Semaphores.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
LGTM (Just an overview of code changes and check that there's no more semphores table after the migration)

**Pull request closed without merging**